### PR TITLE
LFH-52 Sort skills by grade, then category, then title

### PR DESCRIPTION
**Pull request by Learner5200**
This commit ensures that the skills returned from the API are sorted
by grade seniority, then by category title, then by skill title. The
Comparable interface is used to give these model objects responsibility
for their own ordering, and to make it (hopefully) easy to sort
categories or grades independently if needed.

### CODE REVIEW

File: src/test/java/com/loofah/graph/api/integrationTests/DatabaseSeeder.java [hunk 1]
```diff
@@ -20,30 +20,39 @@
 
     public static final List<Skill> SKILLS = Collections.unmodifiableList(Arrays.asList(
```
Comment (Learner5200) [inline src/test/java/com/loofah/graph/api/integrationTests/DatabaseSeeder.java:21]: I've put these in reverse order to allow testing of the sorting behaviour - open to suggestions for neater ways to do that

File: src/test/java/com/loofah/graph/api/models/database/CategoryTest.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    @Test
+    void testSorting() {
+        // Given
+
+        final Category aardvark = getDefaultCategoryBuilder().withTitle("aardvark").build();
+        final Category bear = getDefaultCategoryBuilder().withTitle("bear").build();
+        final Category butterfly = getDefaultCategoryBuilder().withTitle("butterfly").build();
+        final Category cat = getDefaultCategoryBuilder().withTitle("cat").build();
+        final Category nullosaurus = getDefaultCategoryBuilder().withTitle(null).build();
```
Comment (elishkaflint) [inline src/test/java/com/loofah/graph/api/models/database/CategoryTest.java:22]: obsessed 💟

File: src/test/java/com/loofah/graph/api/integrationTests/DatabaseSeeder.java (hunk 1 above)
Comment (elishkaflint) [reply src/test/java/com/loofah/graph/api/integrationTests/DatabaseSeeder.java:21]: i can't think of another way but i think this is ok

**Review by elishkaflint — APPROVED** [review]
This is really good, thanks @Learner5200

File: src/main/java/com/loofah/graph/api/models/database/Category.java [hunk 3]
```diff
@@ -57,6 +62,38 @@ public String getDescription() {
...
+                Objects.equals(description, category.description);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(id, title, description);
+    }
+
+    @Override
+    public int compareTo(@NotNull final Category otherCategory) {
```
Comment (vijamesDD) [inline src/main/java/com/loofah/graph/api/models/database/Category.java:68]: whatre your thoughts on having a little comment on this method (and the same in the other classes) to briefly explain what it does? "naturalOrder" for example isnt clear to me straight away what order that is (assuming alphabetical?)
I've not done loads with ordering though so if you think this is overkill / adding clutter then that is fine with me too

File: src/test/java/com/loofah/graph/api/models/database/CategoryTest.java (hunk 2 above)
Comment (vijamesDD) [reply src/test/java/com/loofah/graph/api/models/database/CategoryTest.java:22]: 🦖 amazing

**Review by vijamesDD — APPROVED** [review]
👏 Really nice ✨ testing coverage as well is 🔝 
I've just added a tiny comment about comments

File: src/main/java/com/loofah/graph/api/models/database/Category.java (hunk 3 above)
Comment (Learner5200) [reply src/main/java/com/loofah/graph/api/models/database/Category.java:68]: Great idea, have added comments to describe each comparison

**Pull request merged**